src/games/strategy/triplea/settings/ValueRange.java

```diff
@@ -0,0 +1,23 @@
+package games.strategy.triplea.settings;
+
+/**
+ * Encapsulates the permissible range for the value of a specific setting.
+ *
+ * <p>
+ * Implementations may choose to track other associated values in addition to the range itself, such as a default value.
+ * </p>
+ */
+public interface ValueRange {
```

Comment:
I think we may be able to avoid the optional return value by having an 'any' value ValueRange that maps to `""`. Likewise, we can us a `String` directly instead of `Optional<String>` by only using `""` to represent the empty case.

With that mentioned, this interface is essentially a supplier. We could potentially keep things simpler by replacing this interface with a `Supplier<String>` or `Supplier<ValueRange>`